telemetry/telemetry.go

```diff
@@ -211,6 +213,57 @@ func isFlagPassed(fs *flag.FlagSet, name string) bool {
 	return found
 }
 
+func iNotifyCertMonitoring(watcher *fsnotify.Watcher, telemetryCfg *TelemetryConfig, serverControlSignal chan<- int, testReadySignal chan<- int) {
+	defer watcher.Close()
+
+	done := make(chan bool)
+	go func() {
+		if testReadySignal != nil { // for testing only
+			testReadySignal <- 0
+		}
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Name != "" {
+					log.V(1).Infof("Inotify watcher has received event: %v", event)
+					if event.Op & fsnotify.Write == fsnotify.Write {
+						log.V(1).Infof("Cert File has been modified: %s", event.Name)
+						serverControlSignal <- 1 // tells gnmi server to stop and restart with new certs
+						done <- true
+						return
+					}
+					if event.Op & fsnotify.Remove == fsnotify.Remove {
```

Comment:
>Remove [](http://example.com/codeflow?start=47&length=6)

Is it possible that a cert file overwrite generate Create event immediate after Remove event? If it is the behavior, you have no way to start.

Reply:
To suicide or not is a critical decision, and I do not see how can we prove 1 second is a good divider for the decision. Let's assume the time is 2 seconds, the gnmi process is already suicided, will it auto restart?

Reply:
Is it a good strategy that if cert is missing, stop gnmi serving any new incoming request, and silently wait for a new cert?

Reply:
Implement logic such that

- If certs are missing when telemetry process is started, we will wait for fsnotify.Create or fsnotify.Write event to check for certs again. Only when certs are present and are good, do we start gnmi server to start serving.

- If the case where certs are overwritten, or are missing as gnmi server is running, we will gracefully stop gnmi server, and then try loading the new certs or wait for certs.

Reply:
I will add change to gracefully stop gnmi server in a different PR.